lib/format_parser.rb

```diff
@@ -58,7 +73,22 @@ def self.parse(io)
         # and examine the file more closely.
       end
     end
-    nil # Nothing matched
+    # Return the array of results if something matched or nil otherwise.
+    results.empty? ? nil : results
```

Comment:
If we go through our parsers and return multiple results, why not turn this into a `map`?

Reply:
Do you any advantage in returning the following map versus the current array?

```ruby
{
  mp4: <FormatParser::Video format=:mp4 ... >,
  mp3: <FormatParser::Audio format=:mp3 ... >, 
  mov: <FormatParser::Video format=:mov ... >
}
```

For sure there are faster lookups for formats, but does that make sense when we're planning to add a filter based upon formats? Also, in case that the keys were different, let's say based on natures, wouldn't it be overdoing? The previous principle applies.

I see some overlapping, therefore I'd pass on this one.

Reply:
That's not what I meant. We are currently using an `.each` do go over the parsers. Since we now preserve multiple return values, we could turn the `.each` into a `.map` and then `.compact` the resulting Array

Reply:
Done!